internal/command/output.go

```
@@ -5,11 +5,118 @@ package command
 
 import (
 	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
 
 	"github.com/hashicorp/jsonapi"
 )
 
-func outputJson(data interface{}) (string, error) {
+type outputMessage struct {
+	// unique name
+	name string
+	// raw value to return to stdout, platform
+	value interface{}
+	// determines if message should be included in stdout. default: true
+	stdOut bool
+	// determines if message should be sent to platform. default: true
+	platformOut bool
+	// if the value may contain strings/json that is multiline
+	multiLine bool
+}
+
+func (o *outputMessage) IncludeWithPlatform() bool {
+	return o.platformOut
+}
+
+func (o *outputMessage) String() (sValue string) {
+	switch o.value.(type) {
+	case string:
+		sValue = fmt.Sprintf("%s", o.value)
+	default:
+		reflectVal := reflect.ValueOf(o.value)
+		reflectInd := reflect.Indirect(reflectVal)
+		refType := reflectInd.Type()
+		// if type is not a struct, return as marshaled string
+		if refType.Kind() != reflect.Struct {
+			b, _ := json.Marshal(o.value)
+			sValue = string(b)
+			return
+		}
+
+		// depending on presence of jsonapi struct tags, determine marshaller
+		switch resolveMarshaler(refType) {
```

I'm generally suspicious about ignoring errors like this...